File: practica3_820574_839304/ProdGenerico.java
/*
* Fichero: ProdGenerico.java
* Autores: Alicia Lazaro Huerta y Manel Jorda Puig Rubio
* Fecha: 29/02/2024
*/

public abstract class ProdGenerico extends Inventario{
    protected double peso;

    ProdGenerico(double _volumen, String _nombre, double _peso)
    {
        super(_volumen);
        nombre = _nombre;
        peso = _peso;
    }

    public double peso()
    {
        return peso;
    }

    @Override
    public String toString()
    {
        return nombre;
    }
}

File: practica3_820574_839304/c++/prodgenerico.cc
/*
* Fichero: prodgenerico.cc
* Autores: Alicia Lazaro Huerta y Manel Jorda Puig Rubio
* Fecha: 09/03/2024
*/

#include "prodgenerico.h"

ProdGenerico::ProdGenerico(const string& _nombre, double _volumen, double _peso)
            :Inventario(_volumen), peso(_peso)
{
    name = _nombre;
}

double ProdGenerico::get_peso() const
{
    return peso;
}

void ProdGenerico::aumentar_nivel()
{
    nivel+=1;
}


ostream& operator<<(ostream& os, const ProdGenerico& r)
{
    os << r.get_descripcion();
    return os;
}
